src/Normalizer/InstantArticleContentEntityNormalizer.php

```diff
@@ -202,31 +197,15 @@ protected function entityViewDisplay(ContentEntityInterface $entity, array $cont
    *
    * @return \Facebook\InstantArticles\Elements\InstantArticle
    *   Modified instant article.
+   *
+   * @throws \Drupal\Core\Entity\EntityMalformedException
    */
   public function normalizeCanonicalUrl(InstantArticle $article, ContentEntityInterface $entity) {
     // Set the canonical URL.
     $article->withCanonicalURL($this->entityCanonicalUrl($entity));
     return $article;
   }
 
-  /**
-   * Helper function to compute the canonical URL for a given entity.
-   *
-   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
-   *   Content entity being normalized.
-   *
-   * @return string
-   *   The canonical URL for the given entity.
-   */
-  public function entityCanonicalUrl(ContentEntityInterface $entity) {
-    if ($override = $this->config->get('canonical_url_override')) {
```

Comment:
we could add linkTemplate to entities via [hook_entity_type_alter](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21entity.api.php/function/hook_entity_type_alter/8.5.x) instead of keeping at config. Something like [this](http://cgit.drupalcode.org/drupal/tree/core/modules/settings_tray/settings_tray.module?h=8.5.x#n99)